library/Core/src/Juvix/Core/IR/CheckTerm.hs

```diff
@@ -375,6 +375,21 @@ requireStar ::
 requireStar (IR.VStar j) = pure j
 requireStar ty = Error.throwTC (Error.ShouldBeStar ty)
 
+-- | Given the arity of a type, require all of them to be Star. We expect the
+-- arguments to be written in a right-associative way, e.g: @* -> (* -> *)@.
+-- TODO: check that pi == 0.
+requirePrimStars ::
+  Error.HasThrowTC' IR.T ext primTy primVal m =>
+  Natural ->
+  Typed.ValueT IR.T primTy primVal ->
+  m Core.Universe
+requirePrimStars 0 ty = requireStar ty
+requirePrimStars n ty = do
+  (π, l, r) <- requirePi ty
+  requireZero π -- Existentially quantified variables should be 0.
```

Comment:
```suggestion
```

I don't think dependent function types should necessarily be of usage 0. See https://bentnib.org/quantitative-type-theory.pdf

<img width="351" alt="Screenshot 2021-09-23 at 12 31 09" src="https://user-images.githubusercontent.com/2612856/134499634-f84de19c-be43-4bbb-980b-67f004076b44.png">

Reply:
oh you're right! i got my πs mixed up so i didn't notice that was a problem

Reply:
Hmm I discussed this with Andy at some point. Can't remember exactly, I thought that he pointed out that it should be zero, because these types are dropped during compilation. What do you think @andy-morris?

Reply:
I think you were correct. I removed `requireZero pi`, now we don't have to do any special usage stuff anymore.